server/mailet/rate-limiter-redis/src/main/scala/org/apache/james/rate/limiter/redis/RedisRateLimiterConfiguration.scala

```diff
@@ -0,0 +1,40 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.rate.limiter.redis
+
+import com.google.common.base.Preconditions
+import io.lettuce.core.RedisURI
+import org.apache.commons.configuration2.HierarchicalConfiguration
+import org.apache.commons.configuration2.tree.ImmutableNode
+
+object RedisRateLimiterConfiguration {
+  def from(configuration: HierarchicalConfiguration[ImmutableNode]): RedisRateLimiterConfiguration = {
+    val redisUri: String = configuration.getString("redisURL", null)
+    val redisKeyPrefix: String = configuration.getString("redisKeyPrefix", null)
+    Preconditions.checkNotNull(redisUri)
+    Preconditions.checkNotNull(redisKeyPrefix)
+    from(redisUri, redisKeyPrefix)
+  }
+
+  def from(redisUri: String, redisKeyPrefix: String) : RedisRateLimiterConfiguration =
+    RedisRateLimiterConfiguration(RedisURI.create(redisUri), redisKeyPrefix)
+}
+
+case class RedisRateLimiterConfiguration(redisURI: RedisURI, redisKeyPrefix: String)
```

Comment:
How can we pass a username and password here?

Reply:
username+password+db in `redisURI`
example: `redis://user1:pass1@localhost:6379/1`

Reply:
`val redisURI = RedisURI.create("redis://user1:pass1@localhost:6379/1")`